### PR TITLE
mds: remove superfluous error in StrayManager::advance_delayed()

### PR DESCRIPTION
**Pull request by ukernel**
Fixes: http://tracker.ceph.com/issues/38679
Signed-off-by: "Yan, Zheng" <zyan@redhat.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/mds/StrayManager.cc [hunk 1]
```diff
@@ -369,18 +369,6 @@ void StrayManager::advance_delayed()
       dout(4) << __func__ << ": delayed dentry is now null: " << *dn << dendl;
       continue;
     }
-
-    const bool purging = eval_stray(dn);
```
Comment (batrick) [inline src/mds/StrayManager.cc]: Can we actually remove this line? I think this causes these errors and others: /ceph/teuthology-archive/pdonnell-2019-03-30_05:01:26-fs-wip-pdonnell-testing-20190329.214758-distro-basic-smithi/3788265/teuthology.log

**Comment by ukernel** [comment]
my fault. I shouldn't remove eval_stray()

**Comment by batrick** [comment]
retest this please

**Comment by xiaoxichen** [comment]
@ukernel  we seems hit this in a different way

```
2019-07-10 03:27:51.458 7fc882aac700 10 mds.9.cache.strays  inode is [inode 0xa0001908553 [2,head] ~mds9/stray2/a0001908553 auth v12728687 s=7890291 nl=0 n(v0 rc2019-07-10 02:39:11.984908 b7890291 1=1+0) (iversion lock) caps={217702=pAsLsXsFscr/-@6} | ptrwaiter=0 request=0 lock=0 caps=1 dirtyparent=0 dirty=0 waiter=0 authpin=0 0x55a7342f9100]
2019-07-10 03:27:51.458 7fc882aac700 20 mds.9.cache.strays  caps | leases
2019-07-10 03:27:51.458 7fc882aac700 -1 mds.9.cache.strays Dentry [dentry #0x109/stray2/a0001908553 [2,head] auth (dversion lock) v=12728687 ino=0xa0001908553 state=1342177280 | request=0 lock=0 inodepin=1 dirty=0 authpin=0 0x55a86e584f00] was purgeable but no longer is!
```

the dn is not purgeable due to still have caps on it, the inode is NOT frozen

```
root@cal-rno-mds-new-27:~# ceph daemon --cluster rno_ceph_cal mds.`hostname -s` dump inode 0xa0001908553
{
    "path": "~mds9/stray2/a0001908553",
    "ino": 10995142526291,
    "rdev": 0,
    "ctime": "2019-07-10 02:39:11.984908",
    "btime": "2019-07-09 06:33:06.046917",
    "mode": 33188,
    "uid": 0,
    "gid": 0,
    "nlink": 0,
    "dir_layout": {
        "dir_hash": 0
    },
    "layout": {
        "stripe_unit": 4194304,
        "stripe_count": 1,
        "object_size": 4194304,
        "pool_id": 2,
        "pool_ns": ""
    },
    "old_pools": [],
    "size": 7890291,
    "truncate_seq": 1,
    "truncate_size": 18446744073709551615,
    "truncate_from": 0,
    "truncate_pending": 0,
    "mtime": "2019-07-09 06:34:49.419935",
    "atime": "2019-07-09 06:33:06.046917",
    "time_warp_seq": 0,
    "change_attr": 1,
    "export_pin": -1,
    "client_ranges": [],
    "dirstat": {
        "version": 0,
        "mtime": "0.000000",
        "num_files": 0,
        "num_subdirs": 0
    },
    "rstat": {
        "version": 0,
        "rbytes": 7890291,
        "rfiles": 1,
        "rsubdirs": 0,
        "rsnaps": 0,
        "rctime": "2019-07-10 02:39:11.984908"
    },
    "accounted_rstat": {
        "version": 0,
        "rbytes": 7890291,
        "rfiles": 1,
        "rsubdirs": 0,
        "rsnaps": 0,
        "rctime": "2019-07-10 02:39:11.984908"
    },
    "version": 12728687,
    "file_data_version": 0,
    "xattr_version": 1,
    "backtrace_version": 12728687,
    "stray_prior_path": "ORIGNIAL_PATH_HIDDEN",
    "symlink": "",
    "old_inodes": [],
    "dirfragtree": {
        "splits": []
    },
    "is_auth": true,
    "auth_state": {
        "replicas": {}
    },
    "replica_state": {
        "authority": [
            9,
            -2
        ],
        "replica_nonce": 0
    },
    "auth_pins": 0,
    "is_frozen": false,
    "is_freezing": false,
    "pins": {
        "ptrwaiter": 0,
        "request": 0,
        "lock": 0,
        "caps": 1,
        "dirtyparent": 0,
        "dirty": 0,
        "waiter": 0,
        "authpin": 0
    },
    "nref": 1,
    "versionlock": {
        "gather_set": [],
        "is_leased": false,
        "num_rdlocks": 0,
        "num_wrlocks": 0,
        "num_xlocks": 0,
        "xlock_by": {}
    },
    "authlock": {},
    "linklock": {},
    "dirfragtreelock": {},
    "filelock": {},
    "xattrlock": {},
    "snaplock": {},
    "nestlock": {
        "gather_set": [],
        "is_leased": false,
        "num_rdlocks": 0,
        "num_wrlocks": 0,
        "num_xlocks": 0,
        "xlock_by": {}
    },
    "flocklock": {},
    "policylock": {},
    "states": [
        "auth",
        "notifyref",
        "orphan"
    ],
    "client_caps": [
        {
            "client_id": 217702,
            "pending": "pAsLsXsFscr",
            "issued": "pAsLsXsFscr",
            "wanted": "-",
            "last_sent": 6
        }
    ],
    "loner": -1,
    "want_loner": -1,
    "mds_caps_wanted": []
}
```

**Comment by xiaoxichen** [comment]
client session 
```
      "id": 217702,
        "num_leases": 0,
        "num_caps": 52229,
        "state": "open",
        "request_load_avg": 35,
        "uptime": 1418992.6697613781,
        "replay_requests": 0,
        "completed_requests": 0,
        "reconnecting": false,
        "inst": "client.217702 v1:IP_HIDDEN:0/2393301553",
        "client_metadata": {
            "features": "00000000000001ff",
            "ceph_sha1": "cbff874f9007f1869bfd3821b7e33b2a6ffd4988",
            "ceph_version": "ceph version 13.2.5 (cbff874f9007f1869bfd3821b7e33b2a6ffd4988) mimic (stable)",
            "entity_id": "cal",
            "hostname": "HIDDEN",
            "mount_point": "/var/mnt/kubelet/pods/4fefd78d-8688-11e9-959d-90e2bae93ac0/volumes/kubernetes.io~cephfs/cal-ceph",
            "pid": "32568",
            "root": "/"
        }
    },
```

**Pull request merged**